tests/FakeItEasy.Specs/AssignsOutAndRefParametersSpecs.cs

```diff
@@ -162,6 +162,30 @@ public static void AssignOutAndRefParameters(IHaveARef subject, string outValue)
         }
 
         [Scenario]
+        public static void AssignOutAndRefParameterForNonVoidDelegateWithoutSpecifyingReturns(
+            NonVoidDelegateWithOutAndRefParameters subject, int refValue, int outValue, int result)
+        {
+            "Given a faked delegate with a non-void return type and ref and out parameters"
+                .x(() => subject = A.Fake<NonVoidDelegateWithOutAndRefParameters>());
+
+            "When the faked delegate is configured to assign the out and ref parameters without setting the return value"
+                .x(() => A.CallTo(() => subject(1, ref refValue, out outValue))
+                    .AssignsOutAndRefParameters(43, 100));
+
+            "And I call the faked delegate"
+                .x(() => result = subject(1, ref refValue, out outValue));
+
+            "Then it returns a Dummy value"
+                .x(() => result.Should().Be(A.Dummy<int>()));

```

Comment:
It would be better to test with a reference type, because `A.Dummy<int>()` returns 0, which is the same as if it wasn't actually returning a dummy. With `string` as the return value, we could test that it returns an empty string rather than null.